Wrappers/Python/cil/optimisation/algorithms/Algorithm.py

```diff
@@ -25,31 +25,21 @@
 
 
 class Algorithm:
-    '''Base class for iterative algorithms
+    r"""Base class for iterative algorithms that provides the minimal infrastructure.
 
-      provides the minimal infrastructure.
+    Algorithms are iterables so can be easily run in a for loop. They will stop as soon as the stop criterion is met.
```

Comment:
Before moving to explaining how it can be used, I think it would be valuable to describe what it achives, i.e., elaborate on what an "iterative algorithm" as mentioned in the first line is. Sth on iteratively solving an optimisation problem, introduce later used terms like iteration, iterate, objective (value), loss, (current) solution, stopping criterion, etc.

Reply:
I gave it a go!